tests/integration/cli_test.py

```diff
@@ -155,6 +157,47 @@ def test_build_no_cache_pull(self):
         self.assertNotIn(cache_indicator, output)
         self.assertIn(pull_indicator, output)
 
+    def test_build_failed(self):
+        self.command.base_dir = 'tests/fixtures/simple-failing-dockerfile'
+
+        with self.assertRaises(BuildError):
+            self.command.dispatch(['build', 'simple'], None)
+
+        labels = [
+            "com.docker.compose.test_failing_image=true",
+        ]
+
+        containers = [Container.from_ps(self.project.client, c) for c in self.project.client.containers(
+            all=True,
+            filters={"label": labels})]
+
+        # First remove the containers, because if the assertion fails, tearDown won't be able to clean up
+        container_count = len(containers)
+        for container in containers:
+            container.remove(force=True)
```

Comment:
To be a little more explicit about this (without needing a comment), please use [`self.addCleanup()`](https://docs.python.org/2/library/unittest.html#unittest.TestCase.addCleanup)